tests/test_post_gen_project.py

```diff
@@ -0,0 +1,184 @@
+        cookiecutter_path.mkdir()
+    actual = get_cookiecutter_dir()
+    expected = home_path / ".cookiecutters" / "scikit-package-manuscript"
+    assert expected == actual
+
+
+@pytest.mark.parametrize(
+    "cookiecutters",
+    [
+        # test get_cookiecutter_dir when "scikit-package-manuscript"
+        #   cookiecutter does not exist.
+        # C1: empty ".cookiecutters", expect FileNotFoundError.
+        (),
+        # C2: without "scikit-package-manuscript", expect FileNotFoundError.
+        ("foo", "bar"),
+    ],
+)
+def test_get_cookiecutter_dir_bad(user_filesystem, cookiecutters):
+    home_path = user_filesystem
+    ck_path = home_path / ".cookiecutters"
+    for cookiecutter in cookiecutters:
+        cookiecutter_path = ck_path / cookiecutter
+        cookiecutter_path.mkdir()
+    with pytest.raises(
+        FileNotFoundError, match=r"Couldn't find scikit-package-manuscript,"
+    ):
+        get_cookiecutter_dir(cookiecutter_name="scikit-package-manuscript")
+
+
+@pytest.mark.parametrize(
+    "templates, expected_template",
+    [
+        # test if copy_journal_template_files can find the expected template.
+        # C1: with multiple templates, expect no NotADiectoryError.
+        (["article", "foo", "bar"], "article")
+    ],
+)
+def test_copy_journal_template_files_find_template(
+    user_filesystem, templates, expected_template
+):
+    home_path = user_filesystem
+    skm_path = home_path / ".cookiecutters" / "scikit-package-manuscript"
+    cwd_path = home_path / "cwd_dir"
+    for template in templates:
+        template_path = skm_path / "templates" / template
+        template_path.mkdir(parents=True, exist_ok=True)
+    copy_journal_template_files(expected_template, cwd_path)
+
+
+@pytest.mark.parametrize(
+    "templates, expected_template",
+    [
+        # test copy_journal_template_files when template is not existed.
+        # C1: empty templates, expect NotADirectoryError.
+        ([], "article"),
+        # C2: template does not exist, expect NotADirectoryError.
+        (["foo", "bar"], "article"),
+    ],
+)
+def test_copy_journal_template_files_find_template_bad(
+    user_filesystem, templates, expected_template
+):
+    home_path = user_filesystem
+    skm_path = home_path / ".cookiecutters" / "scikit-package-manuscript"
+    skm_path.mkdir(parents=True, exist_ok=True)
+    cwd_path = home_path / "cwd_dir"
+    for template in templates:
+        template_path = skm_path / "templates" / template
+        template_path.mkdir(parents=True, exist_ok=True)
+
+    with pytest.raises(
+        NotADirectoryError,
+        match=r"Cannot find the provided journal_tamplate: ",
+    ):
+        cwd_path = copy_journal_template_files(expected_template, cwd_path)
+
+
+@pytest.mark.parametrize(
+    "template_files",
+    [
+        # test copy_journal_template_files with different journal
+        #   template structure.
+        # C1: empty template, expect nothing happens.
```

Comment:
I think the successful tests should all be together.  We would rather have two test functions, one for successful and the other unsuccessful outcomes (that raise errors).  Since you are testing that the right files are copied over here, we can just delete the "no NotADirectory` tests.

I would also look into building the whole templates filesystem in the fixture.  If we want to test finding the template empty we would have to delete them, but that is often less work than building it each time.